plugins/catalog/src/components/CatalogTable/columns.tsx

```diff
@@ -82,23 +82,32 @@ export const columnFactories = Object.freeze({
       ),
     };
   },
-  createSpecTargetsColumn(): TableColumn<CatalogTableRow> {
-    return {
-      title: 'Targets',
-      field: 'entity.spec.targets',
-      render: ({ entity }) => (
-        <>
-          {(entity?.spec?.targets || entity?.spec?.target) && (
-            <OverflowTooltip
-              text={(
-                (entity!.spec!.targets as JsonArray) || [entity.spec.target]
-              ).join(', ')}
-              placement="bottom-start"
-            />
-          )}
-        </>
-      ),
-    };
+  createSpecTargetColumns(): TableColumn<CatalogTableRow>[] {
```

Comment:
I'm realizing that this is a breaking API change and I'm not sure how to handle this. :thinking: I'd like some guidance on this.

Reply:
You should revert all your changes, and then add a single `customFilterAndSearch` field onto the `TableColumn` that you return. Like [for example this](https://github.com/backstage/backstage/blob/e7a5501356422c6105700b8d17e25fc6e91c71dd/plugins/gocd/src/components/GoCdBuildsTable/GoCdBuildsTable.tsx#L200)

Reply:
@freben Thank you for the help. I've applied your suggestion and it's much cleaner that way :+1: